packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java

```diff
@@ -121,11 +134,16 @@ public void updateBuilder(@NonNull CaptureRequest.Builder requestBuilder) {
   }
 
   @VisibleForTesting
-  static Size computeBestPreviewSize(int cameraId, ResolutionPreset preset)
+  static Size computeBestPreviewSize(int cameraId, ResolutionPreset preset, CaptureMode captureMode, Size[] availableOutputSizes)
       throws IndexOutOfBoundsException {
+    // Using max resolution for the preview is not a good use of system resources.
+    // Limiting the max resolution used for the preview to 1080p. is a good balance.
```

Comment:
You say it's 1080p but in `getBestAvailableCameraSizeForResolutionPreset`, the high preset is 720p